### PR TITLE
Digitizer mch

### PR DESCRIPTION
**Pull request by mwinn2**
- Correction of MClabelling
- member renaming in digitizer

### CODE REVIEW

**Comment by sawenzel** [comment]
clang format

**Comment by mwinn2** [comment]
Ok, done.

File: Detectors/MUON/MCH/Simulation/src/Digitizer.cxx [hunk 1]
```diff
@@ -81,22 +81,26 @@ void Digitizer::process(const std::vector<Hit> hits, std::vector<Digit>& digits)
...
     int detID = hit.GetDetectorID();
-    processHit(hit, detID, mEventTime, labelIndex);
+    int ndigits = processHit(hit, detID, mEventTime);
+    //TODO need one label per Digit
+    // can use nDigit output of processHit
     MCCompLabel label(hit.GetTrackID(), mEventID, mSrcID);
-    mMCTruthContainer.addElementRandomAccess(labelIndex, label);
-  } //loop over hits
+    for (int i = 0; i < ndigits; ++i) {
+      mTrackLabels.addElementRandomAccess(labelIndex, label);
```
Comment (aphecetche) [inline Detectors/MUON/MCH/Simulation/src/Digitizer.cxx]: @mwinn2 
Coming back this rather confusing business of labels...
After a discussion with Diego (who's implementing the MID counterpart), we kind of concluded that this is wrong. 
First, it's not a labelIndex, but a dataIndex or digitIndex. And it should not be inferred from the mTrackLabels array(s) but from the digit array, in our opinion.
So the association would be instead : 

for (int i=0; i<ndigits;++i) {
  int digitIndex = mDigits.size() - ndigits + i; // -ndigits as digits were already added to mDigits array during processHit, right ?
  mTrackLabels.addElementRandomAccess(digitIndex,label);
}

Does that make sense ?

Comment (mwinn2) [reply Detectors/MUON/MCH/Simulation/src/Digitizer.cxx]: @aphecetche 
Yes, seems correct that the current implementation in the PR is wrong, a funny mixture of what was there before and what needs to be put. Let me cross check tomorrow with other code in O2 that I convince myself that it is indeed as you concluded with Diego.

Comment (mwinn2) [reply Detectors/MUON/MCH/Simulation/src/Digitizer.cxx]: @aphecetche 
ok, putted in.

**Comment by mwinn2** [comment]
@sawenzel : as far as I can see, the remaining problem in the test is not related to the code itself or do I miss something in the log?

**Comment by sawenzel** [comment]
I was just waiting to get a green a O2/o2 again. Will be merged asap now.

**Comment by mwinn2** [comment]
Ok, very good, thanks!

**Pull request merged**